### PR TITLE
Add 'location_t' data structure

### PR DESCRIPTION
**Pull request by torvalds**
Instead of having people treat latitude and longitude as separate
things, just add a 'location_t' data structure that contains both.

Almost all cases want to always act on them together.

This is really just prep-work for adding a few more locations that we
track: I want to add a entry/exit location to each dive (independent of
the dive site) because of how the Garmin Descent gives us the
information (and hopefully, some day, other dive computers too).

Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

**Review by dirkhh — APPROVED** [review]
Nice. Makes things more readable.

**Comment by neolit123** [comment]
:+1: 
seems like the smtk-import tool needs an update too:
https://api.travis-ci.org/v3/job/444161031/log.txt
(errors near the bottom of the log)

**Review by neolit123 — APPROVED** [review]

File: qt-models/divelocationmodel.cpp [hunk 1]
```diff
@@ -123,28 +123,25 @@ bool GPSLocationInformationModel::filterAcceptsRow(int sourceRow, const QModelIn
...
+	location = locationIn;
 	invalidate();
 }
 
-void GPSLocationInformationModel::setCoordinates(degrees_t latitudeIn, degrees_t longitudeIn)
+void GPSLocationInformationModel::setCoordinates(const location_t &locationIn)
 {
-	latitude = latitudeIn;
-	longitude = longitudeIn;
+	location = locationIn;
```
Comment (bstoeger) [inline qt-models/divelocationmodel.cpp]: `location = location` is a noop.

**Review by bstoeger — APPROVED** [review]
lgtm - excuse my erroneous comment.

Intuitively I would pass by value instead of const reference, since both fit into a 64-bit register. But that's an irrelevant detail.

**Comment by torvalds** [comment]
Yeah, I'll fix that nasty smartrak thing, and also add a helper to create locations from floating point latitude/longitude, since there are a number of places that do that (I created one for the special case of the qmlmapdigethelper, but I'll do the general case).

**Comment by torvalds** [comment]
Ok, pushed updated branch. Further slight cleanups, and the smartrak thing hopefully is right now too.

**Pull request merged**